### DISCUSSION TITLE
Would it be possible to use Mitosis for maintaining different frameworks/libs?

### DISCUSSION THREAD

**Discussion by juliomuhlbauer** [chakra-ui/ark#1819] [Ideas]
https://github.com/BuilderIO/mitosis

**Comment by Omikorin** [comment]
Hi, as stated [here](https://ark-ui.canny.io/feature-requests/p/support-for-other-frameworks-via-mitosis), it is not a use case. We use finite state machines as a base logic layer for components. Mitosis is a generator and uses framework-native solutions, which could introduce bugs. Transitions won't be as predictable as with state machines.

**Reply by juliomuhlbauer** [reply]
I understand. But the integration form Zag to the framework level couldn't be mande with Mitosis?

**Reply by Omikorin** [reply]
We need more than state machines e.g. types, `asChild`. Not forgetting that Zag won't work in Svelte until there is support for [those](https://github.com/chakra-ui/zag/discussions/262) things. Even if possible, it means a total rewrite of Ark UI. We have a roadmap and a huge backlog to work on.

**Reply by juliomuhlbauer** [reply]
Get it.
I thought about using Mitosis, because there are a lot of duplicate code on Ark codebase because of framework specifics. That would decrease the number of code on the codebase and make it easier to work with different frameworks. Because there are like more than 5 frameworks (React, Solid, Vue, Qwik, Svelte...). Maintaining the same amount of code to those frameworks it is a lot of work.
Maybe in the future we can propose something like this then.

**Reply by Omikorin** [reply]
I looked into it further. It could be possible to use Mitosis with Zag. The Mitosis README states they will show examples of how to do it. Managing one codebase for all frameworks would be a relief, as you said.

I found one problem. Zag's compatibility with other frameworks will limit this solution. Check [this](https://github.com/chakra-ui/zag/discussions/392) discussion about Qwik or [this](https://github.com/chakra-ui/zag/discussions/262) about Svelte. These cases may be resolved in the future.

Our priority is releasing Ark UI for Vue and then working on Chakra UI. As for now, there isn't time to try other solutions having an established codebase.

You can submit an RFC for your idea at some point :) In the end, it is Chris's decision.
